libflux/flux-core/src/semantic/bootstrap.rs

```diff
@@ -192,137 +106,309 @@ pub fn infer_stdlib() -> Result<(PolyTypeMap, PolyTypeMap, Fresher, Vec<String>,
+        description: String,
+        /// the parameters of the function
+        parameters: Vec<ParameterDoc>,
+        /// the type of the function
+        flux_type: String,
+    }
+
+    /// Implementation block for FunctionDoc to house future "FunctionDod" methods
+    impl FunctionDoc {
+        /// Constructs a new FunctionDoc with the given name, headline, and flux type
+        fn new_with_args(
+            name: String,
+            headline: String,
+            description: String,
+            typ: String,
+        ) -> FunctionDoc {
+            FunctionDoc {
+                name,
+                headline,
+                description,
+                parameters: vec![],
+                flux_type: typ,
+            }
+        }
+    }
+
+    /// ParameterDoc represents the documentation for a single parameter within a function.
+    #[derive(Debug, Serialize, Deserialize)]
+    struct ParameterDoc {
+        /// the name of the parameter
+        name: String,
+        /// the headline of the parameter
+        headline: String,
+        /// the description of the parameter
+        description: Option<String>,
+        /// a boolean indicating if the parameter is required
+        required: bool,
+    }
+
+    /// new stdlib docs function
+    pub fn stdlib_docs(
+        lib: &PolyTypeMap,
+        files: &AstFileMap,
+    ) -> Result<Vec<PackageDoc>, Box<dyn std::error::Error>> {
+        let mut docs = Vec::new();
+        for file in files.values() {
+            let pkg = generate_docs(&lib, file)?;
+            docs.push(pkg);
+        }
+        Ok(docs)
+    }
+
+    // Generates the docs by parsing the sources and checking type inference.
+    fn generate_docs(
+        types: &PolyTypeMap,
+        file: &ast::File,
+    ) -> Result<PackageDoc, Box<dyn std::error::Error>> {
+        // construct the package documentation
+        // use type inference to determine types of all values
+        let mut all_comment = String::new();
+        let members = generate_values(&file, &types)?;
+        if Some(&file.package) != None {
+            all_comment = comments_to_string(&file.package.as_ref().unwrap().base.comments);
+        }
+        let (headline, description) = seperate_description(&all_comment);
+
+        //TODO check if package name exists and if it doesn't throw an error message
+        Ok(PackageDoc {
+            name: file.package.clone().unwrap().name.name,
+            headline,
+            description,
+            members,
+        })
+    }
+
+    // Separates headline from description
+    fn seperate_description(all_comment: &str) -> (String, Option<String>) {
+        let mut headline: String = "".to_string();
+        let mut description = None;
+        let mut reached_end: bool = false;
+        let mut description_text: String = "".to_string();
+        let parser = Parser::new(&all_comment);
+        for event in parser {
+            match event {
+                Event::Text(t) => {
```

Comment:
Right now we are only concerned with these ones because they govern if we populate the headline and description or not. The other events don't play a role in finding the texts which we are parsing for here.

Reply:
So if someone puts more complex markdown in the headline or description, then they get that error there, saying "unsupported event"?

If that's the case, could that instead say "unsupported markdown in documentation comment?" Might be easier to figure out what went wrong.

Reply:
sure, that sounds reasonable. I have changed the message accordingly.